src/sidebar/widget-controller.js

```diff
@@ -245,13 +247,15 @@ module.exports = function WidgetController(
       return;
     }
     annotationUI.clearSelectedAnnotations();
-    loadAnnotations(annotationUI.frames());
+    loadAnnotations();
   });
 
   // Watch anything that may require us to reload annotations.
   $scope.$watch(function () {
-    return annotationUI.frames();
-  }, loadAnnotations);
+    return annotationUI.frames().map(function(frame) {
+      return frame.uri;
+    });
+  }, loadAnnotations, true);
```

Comment:
When a function with multiple arguments takes a boolean, I find it helpful to add a note with the parameter name, to indicate what it the flag is. eg. `loadAnnotations, true /* deep equality */);`